sympy/physics/control/tests/test_lti.py

```diff
@@ -0,0 +1,233 @@
+    tfm1 = TransferFunctionMatrix(
+        Matrix([1 / (a * s)])
+    )
+    tfm2 = TransferFunctionMatrix(
+        Matrix([(b + s) / (a + s)])
+    )
+    expect = TransferFunctionMatrix(
+        Matrix([(a + s + a * s * (b + s)) / (a * s * (a + s))])
+    )
+    assert together(tfm1.parallel(tfm2).G) == together(expect.G)
+
+    H1 = TransferFunctionMatrix(
+        Matrix([7 / (p**9 + 9*p)])
+    )
+    H2 = TransferFunctionMatrix(
+        Matrix([5 / (s + 2)])
+    )
+    H = TransferFunctionMatrix(
+        Matrix([7 / (p**9 + 9*p) + 5 / (s + 2)])
+    )
+    assert together(H1.parallel(H2).G) == together(H.G)
+
+
+def test_TransferFunctionMatrix_negate():
+    G1 = Matrix([s / (s + 1)])
+    tf1 = TransferFunctionMatrix(G1)
+    assert tf1.neg() == TransferFunctionMatrix(
+        Matrix([-s / (s + 1)])
+    )
+
+    G2 = Matrix([[(4*p - 10) / (2*p + 1), 3 / (p + 2)],
+            [1 / (2*p**2 + 5*p + 2), (p + 1) / (p**2 + 4*p + 4)]])
+    tf2 = TransferFunctionMatrix(G2)
+    assert tf2.neg() == TransferFunctionMatrix(
+        Matrix([[(10 - 4*p) / (2*p + 1),  -3 / (p + 2)],
+            [-1 / (2*p**2 + 5*p + 2), (-p - 1) / (p**2 + 4*p + 4)]])
+    )
+
+
+def test_TransferFunctionMatrix_is_proper():
+    G1 = Matrix([2*p / (2 + p**2 - p), p / (1 + p**3)])
+    assert TransferFunctionMatrix(G1).is_proper
+
+    G2 = Matrix([2*s**6/(2 + s**2 - s), (s**4)/(1 + s**2)])
+    assert not TransferFunctionMatrix(G2).is_proper
+
+
+def test_TransferFunctionMatrix_is_strictly_proper():
+    G1 = Matrix([2*s**6 / (2 + s**2 - s), (s**4) / (1 + s**2)])
+    assert not TransferFunctionMatrix(G1).is_strictly_proper
+
+    G2 = Matrix([[(4*p - 10) / (2*p**3 + 1), 3 / (p + 2)],
+            [1 / (2*p**2 + 5*p + 2), (p + 1) / (p**2 + 4*p + 4)]])
+    assert TransferFunctionMatrix(G2).is_strictly_proper
+
+
+def test_TransferFunction_construction():
+    tf = TransferFunction(s + 1, s**2 + s + 1)
+    assert tf.num == (s + 1)
+    assert tf.den == (s**2 + s + 1)
+    assert tf.args == (s + 1, s**2 + s + 1)
+
+    tf1 = TransferFunction(s + 4, s - 5)
+    assert tf1.num == (s + 4)
+    assert tf1.den == (s - 5)
+    assert tf1.args == (s + 4, s - 5)
+
+    # using different polynomial variables.
+    tf2 = TransferFunction(p + 3, p**2 - 9)
+    assert tf2.num == (p + 3)
+    assert tf2.den == (p**2 - 9)
+    assert tf2.args == (p + 3, p**2 - 9)
+
+    tf3 = TransferFunction(p**3 + 5*p**2 + 4, p**4 + 3*p + 1)
+    assert tf3.args == (p**3 + 5*p**2 + 4, p**4 + 3*p + 1)
+
+    # no pole-zero cancellation on its own.
+    tf4 = TransferFunction((s + 3)*(s - 1), (s - 1)*(s + 5))
+    assert tf4.den == (s**2 + 4*s - 5)
+    assert tf4.args == (s**2 + 2*s - 3, s**2 + 4*s - 5)
+
+    # purely symbolic polynomials.
+    tf5 = TransferFunction(p, p)
+    assert tf5.num == tf5.den == p
+    assert tf5.args == (p, p)
+
+    tf6 = TransferFunction(5, 6)
+    assert tf6.num == 5
+    assert tf6.den == 6
+    assert tf6.args == (5, 6)
```

Comment:
I think you need at least 1 s for it to be a transfer function. This should raise a value error, no?

Reply:
We can have a `TransferFunction` with numerator and denominator being integers. From python-control:
``` python
>>> from control import TransferFunction
>>> s = TransferFunction.s
>>> G = TransferFunction(5, 6); G

5
-
6

>>> type(_)
<class 'control.xferfcn.TransferFunction'>
```

Reply:
Why would I need to create a transfer function for a constant? Yes, you can do it in control but I'm not sure why you'd want to. This should work and be the preferable way I think:

```
tf = TransferFunction(...)
k = S(5)/6
k*tf
```

k is a constant and if it were a TransferFunction none of the methods associated with TransferFunction would really provide anything useful or work at all.

Reply:
Inverse laplace transform of `1` is dirac delta, so transformation function of constant can model impulse.

Reply:
Ok, will update the unit tests that at least 1 `s` is needed for it to be a TF.

Reply:
Thanks for that reminder!